src/components/frame/v5/pages/AgreementsPage/FilterContext/FilterContext.tsx

```diff
@@ -0,0 +1,188 @@
+import React, {
+  type FC,
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
+
+import { type MotionState } from '~utils/colonyMotions.ts';
+import { formatText } from '~utils/intl.ts';
+
+import {
+  type AgreementsPageFilters,
+  type DateOptions,
+} from '../partials/AgreementsPageFilters/types.ts';
+import { getDateFilter } from '../partials/AgreementsPageFilters/utils.ts';
+
+interface FiltersContextValue {
+  searchFilter: string;
+  setSearchFilter: (searchValue: string) => void;
+  motionStates: MotionState[];
+  activeFilters: AgreementsPageFilters;
+  selectedFiltersCount: number;
+  dateFilters: DateOptions;
+  handleMotionStatesFilterChange: (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => void;
+  handleDateFilterChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleCustomDateFilterChange: (date: [Date | null, Date | null]) => void;
+  handleResetFilters: (category: string) => void;
+}
+
+export const FiltersContext = createContext<FiltersContextValue | undefined>(
+  undefined,
+);
+
+export const FiltersContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [searchFilter, setSearchFilter] = useState('');
+  const [motionStates, setMotionStates] = useState<MotionState[]>([]);
+  const [dateFilters, setDateFilters] = useState<DateOptions>({
+    pastHour: false,
+    pastDay: false,
+    pastWeek: false,
+    pastMonth: false,
+    pastYear: false,
+    custom: undefined,
+  });
+
+  const handleMotionStatesFilterChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const isChecked = event.target.checked;
+      const name = event.target.name as MotionState;
+      const failedStatuses = [
+        'Failed',
+        'FailedNotFinalizable',
+      ] as MotionState[];
```

Comment:
I feel like this should be a matter for filters config, e.g. if you could set up a mapping between motion states and arrays of states they should select, it would be much easier to see what value is actually being set without digging through multiple layers of code. Getting rid of code handling specific edge cases is always nice.

```tsx
const MOTION_FILTERS = [
  [MotionState.Failed]: [MotionState.Failed, MotionState.FailedNotFinalizable],
  [MotionState.Passed]: [MotionState.Passed]
  // ...
]
```